stage1/init/common/mount.go

```diff
@@ -108,34 +141,48 @@ func GenerateMounts(ra *schema.RuntimeApp, volumes map[types.ACName]types.Volume
 				GID:  &defaultGID,
 			}
 
-			dockerImplicit := convertedFromDocker(imageManifest)
 			log.Printf("warning: no volume specified for mount point %q, implicitly creating an \"empty\" volume. This volume will be removed when the pod is garbage-collected.", mp.Name)
-			if dockerImplicit {
+			if convertedFromDocker {
 				log.Printf("Docker converted image, initializing implicit volume with data contained at the mount point %q.", mp.Name)
 			}
 
-			volumes[uniqName] = emptyVol
+			vols[uniqName] = emptyVol
 			genMnts = append(genMnts,
 				mountWrapper{
 					Mount: schema.Mount{
 						Volume: uniqName,
 						Path:   mp.Path,
 					},
-					DockerImplicit: dockerImplicit,
+					Volume:         emptyVol,
+					ReadOnly:       mp.ReadOnly,
+					DockerImplicit: convertedFromDocker,
 				})
 		} else {
+			ro := mp.ReadOnly
+			if vol.ReadOnly != nil {
+				ro = *vol.ReadOnly
+			}
 			genMnts = append(genMnts,
 				mountWrapper{
 					Mount: schema.Mount{
 						Volume: vol.Name,
 						Path:   mp.Path,
 					},
+					Volume:         vol,
+					ReadOnly:       ro,
 					DockerImplicit: false,
 				})
 		}
 	}
 
-	return genMnts
+	for _, gm := range genMnts {
+		if gm.Volume.Name == "" {
+			log.Print(gm)
+			panic("gm.volume.name is empty!")
```

Comment:
Oops, this is debugging code that slipped in! I thought, "surely I won't forget to remove the panic!"